app/src/main/kotlin/com/simplemobiletools/gallery/pro/adapters/MediaAdapter.kt

```diff
@@ -475,10 +475,11 @@ class MediaAdapter(
             }
 
             media.removeAll(removeMedia)
-            listener?.tryDeleteFiles(fileDirItems)
+            listener?.tryDeleteFiles(fileDirItems){
+                removeSelectedItems(positions)
+                currentMediaHash = media.hashCode()
+            }
```

Comment:
we should check the new SDK 30 dialog above these lines at handleSAFDialog too, to not do anything before it is handled properly